### PR TITLE
Remove error + portablegabi pallets

### PR DESCRIPTION
**Pull request by wischli**
## fixes KILTProtocol/ticket#940 + KILTProtocol/ticket#941
* Removes our "ancient" `error-pallet` which was required before Substrate supported errors natively
  * Convert all of our custom defined errors to Substrate error type
* Removes our `portablegabi-pallet` 🥲 because this ZKP did not turn out to be as practicable as we hoped (Note: After the KILT launch we will implement another more practicable ZKP)

## How to test:
Check CI or 
```
cargo test --all --all-features
```

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: pallets/attestation/src/lib.rs [hunk 1]
```diff
@@ -63,59 +83,43 @@ decl_module! {
 		pub fn add(origin, claim_hash: T::Hash, ctype_hash: T::Hash, delegation_id: Option<T::DelegationNodeId>) -> DispatchResult {
 			// origin of the transaction needs to be a signed sender account
 			let sender = ensure_signed(origin)?;
+
 			// check if the CTYPE exists
-			if !<ctype::CTYPEs<T>>::contains_key(ctype_hash) {
-				return Self::error(<ctype::Module<T>>::ERROR_CTYPE_NOT_FOUND);
-			}
+			ensure!(<ctype::CTYPEs<T>>::contains_key(ctype_hash), ctype::Error::<T>::NotFound);
```
Comment (weichweich) [inline pallets/attestation/src/lib.rs:88]: love it!

File: pallets/delegation/src/lib.rs [hunk 2]
```diff
@@ -169,53 +186,46 @@ decl_module! {
...
-						parent_id,
-						delegate.clone(),
-						permissions,
-					));
-					// add child to tree structure
-					Self::add_child(delegation_id, parent_id);
-				}
+				ensure!(parent_node.owner.eq(&sender), Error::<T>::NotOwnerOfParent);
+
+				//
```
Comment (weichweich) [inline pallets/delegation/src/lib.rs:208]: I guess that wanted to become a comment? 😁

File: pallets/delegation/src/lib.rs [hunk 3]
```diff
@@ -232,15 +242,13 @@ decl_module! {
...
 			// origin of the transaction needs to be a signed sender account
 			let sender = ensure_signed(origin)?;
+
 			// check if root node exists
-			let mut root = <error::Module<T>>::ok_or_deposit_err(
-				<Root<T>>::get(root_id),
-				Self::ERROR_ROOT_NOT_FOUND
-			)?;
+			let mut root = <Root<T>>::get(root_id).ok_or(Error::<T>::RootNotFound)?;
+
```
Comment (weichweich) [inline pallets/delegation/src/lib.rs:248]: Oh that is so much cleaner now!

File: pallets/delegation/src/lib.rs [hunk 4]
```diff
@@ -260,50 +268,21 @@ decl_module! {
...
 			// owner of the delegation or of one of its parents
-			if !Self::is_delegating(&sender, &delegation_id, max_depth)? {
-				return Self::error(Self::ERROR_NOT_PERMITTED_TO_REVOKE)
-			}
+			ensure!(Self::is_delegating(&sender, &delegation_id, max_depth)?, Error::<T>::UnauthorizedRevocation);
+
 			// revoke the delegation and recursively all of its children
-			Self::revoke(&delegation_id, &sender)?;
-			Ok(())
+			Self::revoke(&delegation_id, &sender)
```
Comment (weichweich) [inline pallets/delegation/src/lib.rs:279]: clippy would have found it, if it wasn't inside this macro. 😞 Looking forward to frame 2.0! 😋

File: pallets/delegation/src/lib.rs [hunk 5]
```diff
@@ -327,39 +306,33 @@ impl<T: Trait> Module<T> {
...
-			)?;
+			// return whether the given account is the owner of the root
+			let root = <Root<T>>::get(delegation_node.root_id).ok_or(Error::<T>::RootNotFound)?;
 			Ok(root.owner.eq(account))
 		}
 	}
 
-	/// Revoke a delegation an all of its children
+	/// Revoke a delegation and all of its children
 	fn revoke(delegation: &T::DelegationNodeId, sender: &T::AccountId) -> DispatchResult {
```
Comment (weichweich) [inline pallets/delegation/src/lib.rs:331]: this is done recursively! 

[call to revoke in revoke_children](https://github.com/KILTprotocol/mashnet-node/pull/112/files#diff-c6838ae4ac2895550be01595658a64a2570e6a022ec8e62a23ad3ac62f8da94dR356)
[call to revoke_children in revoke](https://github.com/KILTprotocol/mashnet-node/pull/112/files#diff-c6838ae4ac2895550be01595658a64a2570e6a022ec8e62a23ad3ac62f8da94dR344)

tracked in KILTprotocol/ticket/issues/1068

**Review by weichweich — APPROVED** [review]
LGTM

Just a stray `//` waiting for some text or to be removed.

The delegation issue can be fixed in a separate PR.

File: pallets/delegation/src/lib.rs (hunk 5 above)
Comment (wischli) [reply pallets/delegation/src/lib.rs]: Good! I added a `FIXME` in 195109b.

File: pallets/delegation/src/lib.rs (hunk 2 above)
Comment (wischli) [reply pallets/delegation/src/lib.rs]: True! Fixed in 19510b 🙃

**Pull request merged**